### ISSUE TITLE
JSpecify mode: false positive passing null to `Consumer<@Nullable V>` obtained through a wildcard receiver

### ISSUE THREAD

**Issue by vlsi** [uber/NullAway#1847]
With JSpecify mode on, `holder.sink().accept(null)`, where `sink()` returns `Consumer<@Nullable V>`, is reported as `passing @Nullable parameter 'null' where @NonNull is required` when `holder` is `Holder<?>`, `Holder<? extends Object>`, or `Holder<? super String>`. The same call through `Holder<String>` is not reported.

**Versions:** NullAway `master` at de6f2e6a (0.14.2-SNAPSHOT), and the 0.14.1 release, which reports the same three `accept(null)` lines. Error Prone 2.50.0, JDK 21.0.9. Flags: `-XepOpt:NullAway:JSpecifyMode=true -XDaddTypeAnnotationsToSymbol=true -XepOpt:NullAway:JSpecifyExperimental=true -XepOpt:NullAway:OnlyNullMarked=true`.

**Code sample.** I ran it as a test in `nullaway/src/test/java/com/uber/nullaway/jspecify/`, with the helper `WildcardTests` uses: `makeTestHelperWithArgs(JSpecifyJavacConfig.withJSpecifyModeArgs(Arrays.asList("-XepOpt:NullAway:OnlyNullMarked=true")))`, and no `// BUG:` lines, so the test prints every diagnostic.

```java
import java.util.List;
import java.util.function.Consumer;
import org.jspecify.annotations.NullMarked;
import org.jspecify.annotations.Nullable;
@NullMarked
class Test {
  interface Holder<V extends @Nullable Object> {
    Consumer<@Nullable V> sink();
    List<@Nullable V> list();
  }
  static void takesNonNull(Object value) {}
  static void concrete(Holder<String> holder) {
    holder.sink().accept(null);
    takesNonNull(holder.list().get(0));
  }
  static void unbounded(Holder<?> holder) {
    holder.sink().accept(null);
    takesNonNull(holder.list().get(0));
  }
  static void extendsBounded(Holder<? extends Object> holder) {
    holder.sink().accept(null);
    takesNonNull(holder.list().get(0));
  }
  static void superBounded(Holder<? super String> holder) {
    holder.sink().accept(null);
    takesNonNull(holder.list().get(0));
  }
}
```

**Expected:** no diagnostic on any `holder.sink().accept(null)` line (13, 17, 21, 25). The parameter type of `accept` is `@Nullable V` whatever `V` is, and NullAway already treats it that way for `Holder<String>` (line 13). The `takesNonNull(holder.list().get(0))` lines (14, 18, 22, 26) are correctly reported and are there for comparison: through the same wildcard receivers, `@Nullable V` in a return type stays nullable.

**Actual:**

```text
/Test.java:14: warning: [NullAway] passing @Nullable parameter 'holder.list().get(0)' where @NonNull is required
    takesNonNull(holder.list().get(0));
                                  ^
    (see http://t.uber.com/nullaway )

/Test.java:17: warning: [NullAway] passing @Nullable parameter 'null' where @NonNull is required
    holder.sink().accept(null);
                         ^
    (see http://t.uber.com/nullaway )

/Test.java:18: warning: [NullAway] passing @Nullable parameter 'holder.list().get(0)' where @NonNull is required
    takesNonNull(holder.list().get(0));
                                  ^
    (see http://t.uber.com/nullaway )

/Test.java:21: warning: [NullAway] passing @Nullable parameter 'null' where @NonNull is required
    holder.sink().accept(null);
                         ^
    (see http://t.uber.com/nullaway )

/Test.java:22: warning: [NullAway] passing @Nullable parameter 'holder.list().get(0)' where @NonNull is required
    takesNonNull(holder.list().get(0));
                                  ^
    (see http://t.uber.com/nullaway )

/Test.java:25: warning: [NullAway] passing @Nullable parameter 'null' where @NonNull is required
    holder.sink().accept(null);
                         ^
    (see http://t.uber.com/nullaway )

/Test.java:26: warning: [NullAway] passing @Nullable parameter 'holder.list().get(0)' where @NonNull is required
    takesNonNull(holder.list().get(0));
                                  ^
    (see http://t.uber.com/nullaway )
```

Assisted-by: Claude Code (claude-opus-5)
